app/controllers/TransferAgreementController.scala

```diff
@@ -39,8 +43,29 @@ class TransferAgreementController @Inject()(val controllerComponents: SecurityCo
     )(TransferAgreementData.apply)(TransferAgreementData.unapply)
   )
 
-  def transferAgreement(consignmentId: Long): Action[AnyContent] = secureAction { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options))
+  private val getConsignmentClient = graphqlConfiguration.getClient[getConsignment.Data, getConsignment.Variables]()
+
+  private def getConsignmentDetails(request: Request[AnyContent],
```

Comment:
That's a good point, I'll start working on extracting it out into a service 👌

Reply:
To add to this, I've seen a few projects (not by TNA) that put the services in another folder rather than the same folder as the controllers. Have we got any conventions about this or shall I just use my own judgement? 😊

Reply:
In Java and Scala, packages are the same as folders, so if you put it into a separate `services` package then it will be in a separate folder. 👍

Reply:
Ah awesome, thanks 😊